addons/zoom-slider/userscript.js

```diff
@@ -0,0 +1,78 @@
+export default async function ({ addon, msg }) {
+  const addSlider = async () => {
+    if (addon.tab.editorMode !== "editor" || addon.self.disabled) {
+      return;
+    }
+
+    const zoom = await addon.tab.waitForElement(".blocklyZoom");
+    const blockly = await addon.tab.traps.getBlockly();
+
+    const foreignObject = document.createElementNS("http://www.w3.org/2000/svg", "foreignObject");
+    foreignObject.setAttribute("height", "500");
+    foreignObject.setAttribute("width", "120");
+    foreignObject.setAttribute("x", "-126");
+    foreignObject.setAttribute("y", "79");
+    const wrapperDiv = document.createElement("div");
+    wrapperDiv.classList.add("sa-zoom-slider-wrapper");
+    const slider = document.createElement("input");
+    slider.type = "range";
+    slider.min = blockly.mainWorkspace.options.zoomOptions.minScale;
+    slider.max = blockly.mainWorkspace.options.zoomOptions.maxScale;
+    slider.step = "0.01";
+    slider.value = blockly.mainWorkspace.scale;
+    slider.classList.add("sa-zoom-slider");
+    const count = document.createElement("input");
+    count.className = addon.tab.scratchClass("input_input-form", "input_input-small", {
+      others: "sa-zoom-slider-count",
+    });
+    count.value = percentScale(blockly.mainWorkspace.scale);
+    wrapperDiv.append(count, slider);
+    foreignObject.append(wrapperDiv);
+    zoom.append(foreignObject);
+
+    slider.addEventListener("mousedown", (e) => {
+      e.stopPropagation();
+    });
+    slider.addEventListener("input", async () => {
+      blockly.mainWorkspace.setScale(slider.value);
```

Comment:
I fixed the unnecessary repeated zoom level updating which means the context menu bug is now fixed, but the zoom jitter is still there because it's still rounding off.

Reply:
What do you mean by zoom jitter? I don't think I notice anything like that when using it.

Reply:
That seems to have also been fixed.

Reply:
Also good